kratos/utilities/entities_utilities.cpp

```diff
@@ -176,24 +177,16 @@ void EntitityIdentifier<TEntity>::GenerateTemplatedTypes(const std::string& rNam
     // Check the type of entity considered
     const std::string entity_type_name = GetEntityTypeName();
 
+    // Prepare regex 
+    const std::string replace_dimension = StringUtilities::ReplaceAllSubstrings(rName, "#D", "(2D|3D)");
+    const std::string replace_number_of_nodes = StringUtilities::ReplaceAllSubstrings(replace_dimension, "#N", "[0-9]+N");
```

Comment:
I am not into regex (@roigcarlo could you take a look into this?) but as i understand you are replacing the # by a number in the range 0,9

you may have elements with 27 nodes in 3D ... but for example there are no elements with 26 ...

Reply:
`[0-9]+N` means "at least one number followed by an N", which can be more than one. You could use `\d+N`as well.

if you want to restrict to specific numbers you could put them in a list, but that kills the purpose of the regex.

Reply:
I would work with any integer

Reply:
ok, after the didactic explanation of @roigcarlo in my office i understood the code.

i post here the explanation for future use:

~~~cpp
Surface#D#N //this is the input

replace_dimension = "Surface(2D|3D)#N" //after the first substitution
replace_number_of_nodes = "Surface(2D|3D)[0-9]+N" //after the second substitution

std::regex pattern(replace_number_of_nodes); //we will look for a regular expression as in "replace_number_of_nodes".
if we find one we add it to the types
~~~

Reply:
~~~c++
#include <iostream>
#include <regex>

int main() {
    std::vector<std::string> strings = {
        "Element2D42N",
        "Element3D123N",
        "Element2D987N",
        "Element3D5N",
        "InvalidString"
    };

    std::regex pattern("Element(2D|3D)[0-9]+N");

    for (const auto& str : strings) {
        if (std::regex_match(str, pattern)) {
            std::cout << str << " matches the pattern." << std::endl;
        } else {
            std::cout << str << " does not match the pattern." << std::endl;
        }
    }

    return 0;
}
~~~

Will return:

~~~sh
Element2D42N matches the pattern.
Element3D123N matches the pattern.
Element2D987N matches the pattern.
Element3D5N matches the pattern.
InvalidString does not match the pattern.
~~~

Reply:
I saw your self explanation later, Okay